src/TestingFramework/TestCase/FunctionalTestCase.php

```diff
@@ -157,52 +144,50 @@
     /**
      * Private utility class used in setUp() and tearDown(). Do NOT use in test cases!
      *
-     * @var FunctionalTestCaseBootstrapUtility
+     * @var AbstractTestSystem
      */
-    private $bootstrapUtility = null;
+    private $testSystem = null;
 
     /**
-     * Calculate a "unique" identifier for the test database and the
-     * instance patch based on the given test case class name.
+     * Setup creates a test instance and database
      *
-     * @return string
+     * This method has to be called with parent::setUp() in your test cases
+     *
+     * @return void
      */
-    protected function getInstanceIdentifier()
+    protected function setUp()
     {
-        return FunctionalTestCaseBootstrapUtility::getInstanceIdentifier(get_class($this));
+        if (!defined('ORIGINAL_ROOT')) {
+            $this->markTestSkipped('Functional tests must be called through phpunit on CLI');
+        }
+        $this->testSystem = TestSystemFactory::getInstance(get_class($this));
```

Comment:
Hm, but wouldn't that mean that there is a dependency between the class name and the testSystem instance?